elastic/apm-server
# apmbench: Replace -warmup-events for -warmup-time

## Motivation/summary

Removes the `-warmup-events` flag and the complicated logic required to
handle timeouts it in favor of a new flag `-warmup-time` which will send
events for the specified duration. This strategy falls more in line with
the intended usage for a warmup phase, where a user wants to warm up the
target server for an amount of time using the specified agents and does
not care much about how many events are sent for that warmup phase.

Also re-formats some unformatted tf files, and adds a new make benchmark
target which auto-tunes the number of agents from the APM Server size.

Last, it changes apmbench to not return an error when the warm-up phase
returns an error. Warm-ups are best-effort and shouldn't impede the
benchmarks from running.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~

## How to test these changes

Run `apmbench` with `-warmup-time` and see if it warms up the APM Server for the set time.

## Related issues

Part of #8775
